### PR TITLE
Create contribution docs

### PR DESCRIPTION
**Pull request by roaldarbol**
I thought this PR might be a good place to discuss how users can best contribute to the `mystmd` projects.

# 1. Centralised or decentralised
Within `executablebooks` there are a few different places to find information, e.g. [within the Community Guide](https://executablebooks.org/en/latest/contribute/) and the [Development Guide](https://compass.executablebooks.org/en/latest/development/conventions.html). I must admit I had never found them before having them pointed out as `mystmd` has its own website. So it begs the question: Should we simply point there or have additional docs for `myst`-related projects? Personally I'd lean towards having docs here too.

# 2. Channels of communication
Which are the main channels of communication, who are the main devs, what are their domains (so people know who to ping when needed).

EDIT choldgraf: I'd say:

- **For chat and real-time conversations**: [The MyST community Discord server](https://discord.gg/VGb2TdrH).
- **For discussion and work planning**: Issues in the `mystmd` repositories.

# 3. When should users create issues?
First of all, I think it's always imperative to make clear that there's no *wrong* way to contribute. As the development guide states:

> These are guidelines, not rules 💡
> 
> This page is meant to help you make your contribution as efficient and helpful as possible, not to lay down strict rules that must be followed at all times. We think these are reasonable patterns to follow, and the EBP tries to follow them as much as it can. But if you prefer to do things otherwise, that is usually just fine 👍.

A good reproducible issue requires substantial effort to create. As there's a chance it's a simple fix or duplicate issue I see two possible ways of going about things if in doubt:

- Ask in Discord channel first for confirmation, then create issue if needed. 
	-   This approach requires a member of the team to follow up, and so is contingent on someone else's actions which can slow down the process. However, it may remove some unnecessary work for the contributor.
- Create issue even if in doubt.
	- Good issues takes time, so there could be waster time on the user end. We should allow poor first issues and add a category by default (`awaits triage`). However if it is indeed a new issue, it may need further information from the user, and so is still contingent on response from a dev.
	
# 4. Where to raise issues?
This is mostly a question about which repositories the mystmd.org is supposed to cover. 
As far as I can tell it's:
- [mystmd](https://github.com/executablebooks/mystmd): Questions about the MyST document engine and functionality not related to specific renderers.
- [myst-theme](https://github.com/executablebooks/myst-theme): Questions related to the book or article web themes for MyST.
- [myst-spec](https://github.com/executablebooks/myst-spec): Questions about the markdown syntax for MyST and standardization efforts for MyST functionality.
- [jupyterlab-myst](https://github.com/executablebooks/jupyterlab-myst): Questions about the JupyterLab extension for MyST.
- [MyST Templates](https://github.com/myst-templates): Repositories that contain templates for rendering MyST documents into various outputs like Latex, JATS, typst, and docx.

Can someone add a sentence about which type of issues should go into what repo?

EDIT: choldgraf added a quick explainer to each

- closes https://github.com/executablebooks/mystmd/issues/1014

### CODE REVIEW

**Comment by LecrisUT** [comment]
A typical thing to have here:
- pre-commit used or other style tools. How to set them up or use them
- testing frameworks, how to run tests, how to write them
- means of communication. Main devs and areas of expertise would also be nice to know who to @

Other than that, there are the how-to-write extensions like #987 that it would be good to follow from this

**Comment by roaldarbol** [comment]
@LecrisUT Good points! For lots of the developer stuff there's the executablebooks [development guide](https://compass.executablebooks.org/en/latest/development/conventions.html) - I don't know if there are differences in how the devs of *these* repositories prefer things to be coded. Let's await some feedback on that. ☺️

**Comment by rowanc1** [comment]
Thank you both @roaldarbol and @LecrisUT for your help in adding and thinking about these. This is a current focus that @agoose77 and myself are aiming to improve and we welcome all the help we can get to shape these.

Some of the very initial improvements for documenting the install/test/lint side could be moved from the readme to this file (see #1011) linked, then expanded over time.

Other places to get a hold of us are in [Discord](https://discord.mystmd.org/), Github issues/discussion (scoped to this repo).

**Comment by roaldarbol** [comment]
@rowanc1 Who's in the team? And do you have different areas that you cover each?
In the main executablebooks org, there are also Discussions - would you prefer to not have that one listed too?

**Comment by rowanc1** [comment]
The wider executable books team is here:
https://compass.executablebooks.org/en/latest/team/index.html

Main developers (with some non-exhaustive, specialized areas of expertise) of mystmd are @rowanc1 (general), @fwkoch (exports, cli design, jats), @stevejpurves (jupyter outputs), @agoose77 (execution). Main developers of jupyterlab-myst are: @agoose77 (general), @rowanc1 (to a lesser extent, but I help in the theme areas). Main developers of myst-theme: @rowanc1, @stevejpurves (all things jupyter/thebe especially) and increasingly @agoose77.

We have both the main discussion and the one in this repository (@agoose77 you turned that on a while back I think?!) A question to ask is if should we have both especially as the projects are designed to come together? Centralizing the discussions might be a good idea.

**Comment by roaldarbol** [comment]
With the last commits `CONTRIBUTING.md` can live in the base repo and still be included in the docs.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+This page contains pointers and links to help you contribute to this project.
+
+## Our team compass
+
+The {external+tc:doc}`Team Compass <index>` is a source of truth for our team structure and policy.
+It also has a lot of information about how to contribute.
+
+## Code of conduct
+
+We expect all contributors to this project to [](inv:tc#code-of-conduct).
```
Comment (LecrisUT) [inline CONTRIBUTING.md:12]: In order to better display these links, I suggest to use reference-style link, i.e.
```markdown
[Text to display][link-id]

[Link-id]: /path/to/object
```
Combined with `include` with `start-after` `end-before`, you can use html comments like `<!-- MYST-START -->` to separate the text part from the actual link and have `inv:` style links in the html document and actual links in the Github rendered view.

Example: https://github.com/LecrisUT/CMake-Template/blob/main/README.md?plain=1

Comment (agoose77) [reply CONTRIBUTING.md:12]: This is a good idea @LecrisUT. We have an [outstanding issue](https://github.com/executablebooks/mystmd/issues/851) about implementing the link scheme resolution for intersphinx, which right now changes between MySTMD and myst-parser

**Comment by agoose77** [comment]
In general I slightly dislike Discussions; they feel hard to get visibility over, and end up attracting content that should be in one of the sub-repos. But, I think they do provide a useful catch-all for people unsure where to post, so it's probably just the UX that is sub-optimal. 

I don't recall turning on any Discussions myself (I wouldn't have been so bold!) unless you mean the jupyterlab-myst discussions, but I do think there's merit in having EB Discussion (for all our EB tools, catching those that don't end up in e.g. myst-parser), OR having that in `executablebooks/jupyter-book` and closing the EB discussions.

As for MyST, we might need separate discussions for jupyterlab-myst. I can see the benefit in closing the JLab discussions and just having them on myst's org/repo.

**Comment by LecrisUT** [comment]
From the viewpoint of the user, it would depend on how well integrated each component is. Right now, I don't think it's very necessary to have an organization-level discussions, but if integrated services like having a RTD builder for `myst` or integrating `jupyter` to support myst markdown by default, then I think it would be useful, since the regular user would not be familiar with the infrastructure to know where to submit the issue. Right now the only contender for such issues are the theme issues.

As for the workflow, I think the top-level discussions would always be moved to the appropriate repository's discussion/issue after triaging.

Personally, I also dislike the implementation of GH discusssions, but I don't think there are other tools that don't create an entry-barrier for the user to participate. Gitlab has organization level issues, which is more appropriate for most cases, but I think gitlab has more negatives to justify it. Another approach would be to have a `general` repository for catch-all issues that can be later triaged.

**Review by choldgraf — COMMENTED** [review]
How do folks feel about this PR? It may not be totally comprehensive but it feels like a good step in the right direction. Should we try to merge it ASAP and iterate from there? Or incorporate some of the discussions/info provided in this PR?

@roaldarbol what do you prefer to do?

(ps: I also added some answers to a few questions in the top comment in case that helps move things along)

**Review by choldgraf — APPROVED** [review]
I went ahead and made a few changes to the PR based on the discussions in here, with the goal of getting this ready for merge. I think it's a clear improvement on what we've got now, so I'd be happy merging it as-is. Happy to iterate if people want to improve it further, but I'd suggest that we try and do so quickly so that we don't lose the momentum from this PR. What do you think @rowanc1 or @agoose77 ?

**Comment by rowanc1** [comment]
I have updated the links to basic markdown as this will also be in the main github view. If we want to move it completely into the docs folder, we can bring those back.

Agree @choldgraf, lets get this in, we can always iterate. @agoose77 or others if you have further comments feel free to review here or open another PR.

Thanks again @roaldarbol and @LecrisUT for prompting this and the discussions around it.

🚀

**Pull request merged**